go/parquet/internal/bmi/bmi_arm64.go

```diff
@@ -14,40 +14,38 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+//go:build !noasm
 // +build !noasm
 
 package bmi
 
 import (
 	"os"
 	"strings"
-)
-import (
-	"golang.org/x/sys/cpu"
+
+	"github.com/klauspost/cpuid/v2"
 )
 
 func init() {
```

Comment:
This code looks duplicated in two different source files, is it me?

Reply:
the reason why the code is duplicated is because this runs at image startup (via the `init` function) which means if this package isn't imported, then this function never runs. The bottom of the function sets the appropriate function pointer based on whether or not the processor actually supports things or not, and this package is a low level utility that doesn't import anything else so there isn't a shared place I could put this initialization between the two places.

Reply:
I switched this around so that it doens't need to be duplicated anymore.